lib/domain/model/contact/contact.dart

```diff
@@ -24,4 +30,52 @@ class Contact extends Equatable {
   @override
   List<Object?> get props =>
       [email, displayName, matrixId, phoneNumber, status];
+
+  Contact copyWith({
+    String? email,
+    String? displayName,
+    String? matrixId,
+    String? phoneNumber,
+    ContactStatus? status,
+  }) {
+    return Contact(
+      email: email ?? this.email,
+      displayName: displayName ?? this.displayName,
+      matrixId: matrixId ?? this.matrixId,
+      phoneNumber: phoneNumber ?? this.phoneNumber,
+      status: status ?? this.status,
+    );
+  }
+
+  String? get thirdPartyId {
+    if (phoneNumber != null) {
+      return phoneNumber!.msisdnSanitizer();
+    }
+    return email;
+  }
+
+  String? get thirdPartyIdType {
+    if (phoneNumber != null) {
+      return 'msisdn';
+    }
+    return email != null ? 'email' : null;
+  }
```

Comment:
Why don't you make an enum for `thirdPartyIdType`

Reply:
done